test/conf/configuration_test.go

```diff
@@ -0,0 +1,193 @@
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !val {
+		t.Fatalf("Expected %t, got %t", true, val)
+	}
+	err = db.SetConfigValue(ctx, conf.ConfDryRun, "false")
+	if err != nil {
+		t.Fatal(err)
+	}
+	val, err = conf.DryRun(ctx, db)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if val {
+		t.Fatalf("Expected %t, got %t", false, val)
+	}
+	err = db.SetConfigValue(ctx, conf.ConfDryRun, "THIS IS NOT A VALID BOOLEAN!")
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = conf.DryRun(ctx, db)
+	if err == nil {
+		t.Fatalf("Expected this to fail.")
+	}
+}
+
+// testMinPinners ensures MinPinners functions correctly.
+func testMinPinners(t *testing.T, db *database.DB) {
+	ctx := context.Background()
+	val, err := conf.MinPinners(ctx, db)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Make sure we get the default value.
+	if val != 1 {
+		t.Fatalf("Expected %d, got %d", 1, val)
+	}
+	// Set min pinners to a valid value [1;10].
+	mp := 1 + fastrand.Intn(9)
+	err = db.SetConfigValue(ctx, conf.ConfMinPinners, strconv.Itoa(mp))
+	if err != nil {
+		t.Fatal(err)
+	}
+	val, err = conf.MinPinners(ctx, db)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if val != mp {
+		t.Fatalf("Expected %d, got %d", mp, val)
+	}
+
+	// expectBuildCritical fetches the value of MinPinners and fails the test if
+	// it doesn't detect a panic.
+	expectBuildCritical := func() {
+		err = func() (err error) {
+			defer func() {
+				if e := recover(); e != nil {
+					err = nil
+				} else {
+					err = errors.New("expected a panic but didn't get it")
+				}
+			}()
+			_, _ = conf.MinPinners(ctx, db)
+			return
+		}()
+		if err != nil {
+			t.Fatalf("Expected a build.Critical but didn't get it.")
+		}
+	}
+
+	// Set to less than the minimum.
+	err = db.SetConfigValue(ctx, conf.ConfMinPinners, "0")
+	if err != nil {
+		t.Fatal(err)
+	}
+	expectBuildCritical()
+	// Set to less than the minimum and negative.
+	err = db.SetConfigValue(ctx, conf.ConfMinPinners, "-5")
+	if err != nil {
+		t.Fatal(err)
+	}
+	expectBuildCritical() // Set to more than the maximum.
+	err = db.SetConfigValue(ctx, conf.ConfMinPinners, "100")
+	if err != nil {
+		t.Fatal(err)
+	}
+	expectBuildCritical()
+}
+
+// testNextScan ensures NextScan and SetNextScan function correctly.
+func testNextScan(t *testing.T, db *database.DB) {
+	ctx := context.Background()
+	logger := test.NewDiscardLogger()
+	// Ensure that there is no value in the DB.
+	_, err := db.ConfigValue(ctx, conf.ConfNextScan)
+	if !errors.Contains(err, mongo.ErrNoDocuments) {
+		t.Fatalf("Expected '%s', got '%s'", mongo.ErrNoDocuments, err)
+	}
+	// Get the NextScan value, expect this to set the default offset.
+	t0, err := conf.NextScan(ctx, db, logger)
+	if err != nil {
+		t.Fatal(err)
+	}
+	now := time.Now().UTC()
+	if t0.After(now.Add(time.Second)) || t0.Before(now.Add(-1*time.Second)) {
+		t.Fatalf("Expected '%s', got '%s'", now.Add(conf.DefaultNextScanOffset), t0)
+	}
+	// Try to set an invalid value.
+	err = conf.SetNextScan(ctx, db, time.Now().UTC().Add(-1*time.Hour))
```

Comment:
Yes, it will. But this one is really obvious when reading the test. Do you think `time.Now().UTC()` gives us anything extra? If we want to have the test value closer to the limit maybe `time.Now().UTC().Add(SleepBetweenChecksForScan - time.Millisecond)` will be better, as it cuts it right to the limit?

Reply:
nah fine to resolve